logic/resolvable/Unifier.java

```diff
@@ -77,38 +78,41 @@ private Unifier(Map<Identifier, Set<Identifier>> unifier, Requirements requireme
                 }
             }
         }
-        return Optional.of(conceptMap(unifiedMap));
+        return Optional.of(new Pair<>(conceptMap(unifiedMap), new Requirements.Instance(conceptMap.concepts())));
```

Comment:
when forward unifiying, we record the original concept map as `Instance` requirements, which can be provided again in the `unUnify()` stage, and must also be met!